azure-iot-device/azure/iot/device/common/pipeline/config.py

```diff
@@ -26,10 +27,17 @@ def __init__(self, websockets=False, cipher="", proxy_options=None):
         :param cipher: Optional cipher suite(s) for TLS/SSL, as a string in
             "OpenSSL cipher list format" or as a list of cipher suite strings.
         :type cipher: str or list(str)
+        :param proxy_options: Details of proxy configuration
+        :type proxy_options: :class:`azure.iot.device.common.models.ProxyOptions`
         """
         self.websockets = websockets
         self.cipher = self._sanitize_cipher(cipher)
-        self.proxy_options = proxy_options
+
+        # TODO: validate this deeper
```

Comment:
># TODO: validate this deeper [](start = 8, length = 28)

What specifically do you want to validate and why?  Doesn't this violate eafp?  (both the changes from this PR and also any additional validation you want to add)

Reply:
Normally, overly checking something isn't good. However, in this case, there's a lot of weird errors that could happen if this item isn't a ProxyOptions object, since the user provided object gets directly used quite deeply in the stack, and thus we probably want an assurance that it really is a ProxyOptions object.

As far as validating the values in the object itself - you're probably right, we don't need to validate those, so I could remove this TODO

---
In reply to: [393421889](https://github.com/Azure/azure-iot-sdk-python/pull/493#discussion_r393421889) [](ancestors = 393421889)